src/models.rs

```diff
@@ -433,6 +434,14 @@ impl<'a> Uses<'a> {
             Uses::Repository(repo) => repo.git_ref.is_none(),
         }
     }
+
+    pub(crate) fn unpinned_sha(&self) -> bool {
```

Comment:
Bikeshed: `unpinned_sha` is a slightly confusing name for me, maybe this could be `unhashed` if the intended logic is "this `uses:` is either entirely unpinned or, if pinned, does not use a hashed pin"?

Reply:
I think that's a good color of paint, yes. I didn't like the initial name, but I wanted something that suggested similarity between the two. Your suggestion is a big improvement. I'll go with it unless something brilliant comes to mind before I sit down to work on this later today.